frontend/src/modules/Pvt/utils/PvtPlotBuilder.ts

```diff
@@ -0,0 +1,335 @@
+                    patch = { ...patch, [`xaxis${axisIndex}`]: { title: "Pressure [" + xUnit + "]" } };
+                }
+
+                patches.push(patch);
+            }
+        }
+
+        this._figure = makeSubplots({
+            width: size.width,
+            height: size.height,
+            numRows,
+            numCols,
+            sharedXAxes: false,
+            sharedYAxes: false,
+            showGrid: true,
+            margin: { t: 30, b: 40, l: 60, r: 10 },
+            subplotTitles: subplotTitles,
+        });
+
+        for (const patch of patches) {
+            this._figure.updateLayout(patch);
+        }
+    }
+
+    makeTraces(
+        dependentVariables: readonly PressureDependentVariable[],
+        pvtNums: readonly number[],
+        phase: PhaseType,
+        colorBy: ColorBy,
+        colorSet: ColorSet
+    ): void {
+        const figure = this.getFigureAndAssertValidity();
+
+        const tableCollections = this._pvtDataAccessor.getTableCollections();
+
+        this.addLegendTitle(colorBy);
+        const colors = this.makeColorsArray(colorBy, colorSet, pvtNums.length, tableCollections.length);
+
+        let collectionIndex = 0;
+        let pvtNumIndex = 0;
+        for (const tableCollection of tableCollections) {
+            pvtNumIndex = 0;
+            for (const table of tableCollection.tables) {
+                if (pvtNums.includes(table.pvtnum) && phase === table.phase) {
+                    const groupedTracesMaps = new Map<PressureDependentVariable, Map<number, TracePointData[]>>();
+
+                    for (let i = 0; i < table.pressure.length; i++) {
+                        for (const dependentVariable of dependentVariables) {
+                            if (
+                                phase === PhaseType.WATER &&
+                                dependentVariable === PressureDependentVariable.FLUID_RATIO
+                            ) {
+                                continue;
+                            }
+
+                            let dependentVariableValue = 0;
+                            switch (dependentVariable) {
+                                case PressureDependentVariable.FORMATION_VOLUME_FACTOR:
+                                    dependentVariableValue = table.volumefactor[i];
+                                    break;
+                                case PressureDependentVariable.VISCOSITY:
+                                    dependentVariableValue = table.viscosity[i];
+                                    break;
+                                case PressureDependentVariable.DENSITY:
+                                    dependentVariableValue = table.density[i];
+                                    break;
+                                case PressureDependentVariable.FLUID_RATIO:
+                                    dependentVariableValue = table.ratio[i];
+                                    break;
+                            }
```

Comment:
Refactor into private helper function? With if-statements over switch-case? 
Also ensure valid index `i` when accessing element in respective `table` attributes (volumefactor, viscosity, ...).